hudi-aws/src/main/java/org/apache/hudi/aws/credentials/HoodieConfigAWSAssumedRoleCredentialsProvider.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.aws.credentials;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider;
+import org.apache.hudi.common.util.StringUtils;
+import org.apache.hudi.config.HoodieAWSConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Properties;
+
+/**
+ * Credentials provider which fetches AWS access key from Hoodie config.
+ */
+public class HoodieConfigAWSAssumedRoleCredentialsProvider implements AWSCredentialsProvider {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HoodieConfigAWSAssumedRoleCredentialsProvider.class);
+
+  private final STSAssumeRoleSessionCredentialsProvider credentialsProvider;
+
+  public HoodieConfigAWSAssumedRoleCredentialsProvider(Properties props) {
+    if (!validConf(props)) {
+      LOG.debug("AWS role ARN not found in the Hudi configuration.");
+      throw new IllegalArgumentException("AWS role ARN not found in the Hudi configuration.");
```

Comment:
we could skip raising an exception, and let the method `getCredentials` return a None when there is no role ARN provided, in this case the client will try with the next provider in the chain.

Reply:
There is no need to log a debug log.

Reply:
Make it an error message as it is followed by exception throwing.

Reply:
I added it to be aligned with:
https://github.com/apache/hudi/blob/9de7abef5edf04c3c21a914fdaa5a644739d06b0/hudi-aws/src/main/java/org/apache/hudi/aws/credentials/HoodieConfigAWSCredentialsProvider.java#L48

WDYT?

Reply:
@hussein-awala : Can you make it logger.error in both places ?

Reply:
sure

Reply:
This one will be removed as we discussed [here](https://github.com/apache/hudi/pull/9260#discussion_r1393291561); I will change the log level of the other class later in a separate PR.